### PR TITLE
Adds a proof harness for aws_cryptosdk_rsa_encrypt function

### PR DESCRIPTION
**Pull request by feliperodri**
Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>

*Description of changes:*

This PR overrides https://github.com/aws/aws-encryption-sdk-c/pull/437.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: .cbmc-batch/jobs/aws_cryptosdk_sig_verify_finish/aws_cryptosdk_sig_verify_finish_harness.c [hunk 1]
```diff
@@ -43,13 +43,4 @@ void aws_cryptosdk_sig_verify_finish_harness() {
...
     aws_cryptosdk_sig_verify_finish(ctx, signature);
-
-    /* clean up (necessary because we are checking for memory leaks) */
-    free(signature);
-    if (pkey_references > 2) {
-        evp_pkey_unconditional_free(pkey);
-        ec_key_unconditional_free(keypair);  // pkey holds a reference to keypair, have to free that as well
-    } else if (keypair_references > 2) {     // pkey was freed but keypair was not
-        ec_key_unconditional_free(keypair);
-    }
```
Comment (feliperodri) [inline .cbmc-batch/jobs/aws_cryptosdk_sig_verify_finish/aws_cryptosdk_sig_verify_finish_harness.c]: @danielsn should I remove it?

Comment (danielsn) [reply .cbmc-batch/jobs/aws_cryptosdk_sig_verify_finish/aws_cryptosdk_sig_verify_finish_harness.c]: Does it break anything to?

File: .cbmc-batch/source/openssl/objects_override.c [hunk 2]
```diff
@@ -24,8 +24,6 @@
  */
 int OBJ_txt2nid(const char *s) {
     // Currently these are the only values used in the ESDK
-    assert(!s || strcmp(s, "prime256v1") == 0 || strcmp(s, "secp384r1") == 0);
-
```
Comment (feliperodri) [inline .cbmc-batch/source/openssl/objects_override.c]: @danielsn I fixed!

Comment (danielsn) [reply .cbmc-batch/source/openssl/objects_override.c]: thanks

File: .cbmc-batch/jobs/aws_cryptosdk_sig_verify_finish/aws_cryptosdk_sig_verify_finish_harness.c (hunk 1 above)
Comment (feliperodri) [reply .cbmc-batch/jobs/aws_cryptosdk_sig_verify_finish/aws_cryptosdk_sig_verify_finish_harness.c]: Nop

**Pull request merged**